viewing_party/main.py

```diff
@@ -0,0 +1,272 @@
+# return list of dictionary ==> movies that the user hasn't watched
+# similar to part 1 but the reverse reverse!!!
+
+def get_friends_unique_watched(user_data):
+    user_movies = []
+    friends_movies = []
+    friends_unique = []
+
+    for friends in user_data["friends"]:
+        for movie in friends["watched"]:
+            if movie not in user_data["watched"]:
+                friends_movies.append(movie)
+
+    if (len(user_data["watched"])) == 0:
+        for friends in user_data["friends"]:
+            for movie in friends["watched"]:
+                friends_movies.append(movie)
+
+    friends_unique_pt2 = []
+    for i in range(len(friends_movies)):
+        if friends_movies[i] not in friends_movies[i + 1:]:
+            friends_unique_pt2.append(friends_movies[i])
+
+    return friends_unique_pt2
+
+# <-------------- WAVE 4 ------------------ WAVE 4 ------------------------- >
+
+# create a function named get_available_recs, function takes 1 parameter: user_data
+# determine a LIST (with nested dict) of recommended movies
+# movie should be added to the list IF AND ONLY IF:
+# user has not watched it 
+# at least one of the user's friends has watched
+# host of the movie is a service that is in the user's subscriptions
+# return list of recommended movies
+
+def get_available_recs(user_data):
+
+    friends_host = []
+    user_host = (user_data["subscriptions"])
+
+    for friends in user_data["friends"]:
+        for movie in friends["watched"]:
+            if movie["host"] in user_host:
+                friends_host.append(movie)
+
+    friends_host_pt2 = []
+    for i in range(len(friends_host)):
+        if friends_host[i] not in friends_host[i + 1:]:
+            friends_host_pt2.append(friends_host[i])
+
+    return friends_host_pt2
+
+# <-------------- WAVE 5 ------------------ WAVE 5 ------------------------- >
+
+# test wave 5, part 1:
+# create a function get_new_rec_by_genre, function takes 1 parameter: user_data
+# consider the user's most frequently watched genre THEN determine a list of recommended movies
+# movie should only be added to recommended movies list IF AND ONLY IF:
+# the user has not watched it
+# at least one of the user's friends has watched
+# genre of the movie == user's most frequent genre
+# return list of recommended movies
+
+def get_new_rec_by_genre(user_data):
+    recommended_movies = []
+
+# append user's watched titles into new list
+    user_titles = []
+    for details in user_data["watched"]:
+        user_titles.append(details["title"])
+
+# append user's genre into new list
+    user_genre = []
+    for details in user_data["watched"]:
+        user_genre.append(details["genre"])
+
+# create dictionary to determine user's msot frequent genre type
+    user_genre_frequency = {}
+    for type in user_genre:
+        if type not in user_genre_frequency:
+            user_genre_frequency[type] = 1
+        else:
+            user_genre_frequency[type] += 1
```

Comment:
This code is repeated from a previous method. Situations like this are a great chance to practice writing DRY code by calling methods that are already written